docs/syntax.md

```diff
@@ -307,6 +309,27 @@ Supported color codes:
 - `BW` for alternating black and white
 
 
+## Images
+
+Both connectors and cables accept including an image with a caption within their respective nodes.
+
+```yaml
+image:
+  src: <path>        # path to the image file
+  # optional parameters:
+  caption: <str>     # text to display below the image
+  width: <int>       # range: 1~65535; unit: ???
+  height: <int>      # range: 1~65535; unit: ???
```

Comment:
In xxx you mentioned the unit to be `pt`... I'm not too sure. The [GraphViz docs](https://graphviz.org/doc/info/attrs.html#d:image) mention `pt` as the default for SVG files; for bitmaps, they only mention

> The file must contain the image size information. This is usually trivially true for the bitmap formats.

And I wonder if it is then in `px`?

Reply:
How about the simplified version here:
```
image:
  src: <path>        # path to the image file
  # optional parameters:
  caption: <str>     # text to display below the image
  width: <int>       # scale image to width in points
  height: <int>      # scale image to height in points
  # image might get distorted if both width and height are set
```
[More advanced scaling](advanced.md) is also possible, but seldom needed.

I suggest adding the advanced description below to a new `.md` file:

## Advanced Image Scaling

In rare cases when the ordinary image scaling functionality is insufficient, a couple of extra optional image parameters offer extra scaling functionality when combined with the image dimension parameters `width` and `height`:
- `scale: <str>` is default `false` if no dimension is specified, or `true` if only one dimension is specified, or `both` if both dimensions are specified.
- `fixedsize: <bool>` is default `false` when no dimension is specified or if `scale` is specified, and `true` otherwise.
- If `fixedsize` is true and only one dimension is specified, then the other dimension is calculated using the image aspect ratio. If reading the aspect ratio fails, then 1:1 ratio is assumed.

### The effect of `fixedsize` boolean values

- When `false`, any `width` or `height` values are _minimum_ values used to expand the image cell size for more available space, but cell contents or other size demands in the table might expand this cell even more than specified by `width` or `height`.
- When `true`, both `width` and `height` specifiers are required and specify the fixed size of the image cell, distorting any image inside if it don't fit. Any borders are normally drawn around the fixed size, and therefore, WireViz enclose the image cell in an extra table without borders when `fixedsize` is true to keep the borders around the outer non-fixed cell.

### The effect of `scale` string values:

- When `false`, the image is not scaled.
- When `true`, the image is scaled proportionally to fit within the available image cell space.
- When `width`, the image width is expanded (height is normally unchanged) to fill the available image cell space width.
- When `height`, the image height is expanded (width is normally unchanged) to fill the available image cell space height.
- When `both`, both image width and height are expanded independently to fill the available image cell space.

In all cases (except `true`) the image might be distorted when a specified fixed image cell size limits the available space to less than what an unscaled image needs.

In the WireViz diagrams there are no other space demanding cells in the same row, and hence, there are never extra available image cell space height unless a greater image cell `height` also is specified.

### How Graphviz support this image scaling

The connector and cable nodes are rendered using a HTML `<table>` containing an image cell `<td>` with `width`, `height`, and `fixedsize` attributes containing an image `<img>` with `src` and `scale` attributes. See also the [Graphviz doc](https://graphviz.org/doc/info/shapes.html#html), but note that WireViz uses default values as described above.